File: Core/src/main/java/dev/lynith/core/ui/MinecraftScreen.java
package dev.lynith.core.ui;

public abstract class MinecraftScreen {

    public abstract void render(int mouseX, int mouseY, float delta);
    public abstract void init();

    public void closed() {}

    public void mouseClicked(int mouseX, int mouseY, int button) {}
    public void mouseReleased(int mouseX, int mouseY, int button) {}
    public void mouseClickedMoved(int mouseX, int mouseY, int button, long time) {}

    public void keyPressed(char typedChar, int keyCode) {}

    public boolean shouldPauseGame() {
        return true;
    }

}

File: Core/src/main/kotlin/dev/lynith/core/ui/MinecraftScreen.kt
package dev.lynith.core.ui

abstract class MinecraftScreen {


    abstract fun render(mouseX: Int, mouseY: Int, delta: Float)
    abstract fun init()
    fun closed() {}
    fun mouseClicked(mouseX: Int, mouseY: Int, button: Int) {}
    fun mouseReleased(mouseX: Int, mouseY: Int, button: Int) {}
    fun mouseClickedMoved(mouseX: Int, mouseY: Int, button: Int, deltaX: Double, deltaY: Double) {}
    fun keyPressed(typedChar: Char, keyCode: Int) {}
}
